### PR TITLE
automatic filling of Unit parameters

### PR DESCRIPTION
**Pull request by ldrygala**
fixes: https://github.com/scalalandio/chimney/issues/138

### CODE REVIEW

File: chimney/src/test/scala/io/scalaland/chimney/DslSpec.scala [hunk 1]
```diff
@@ -420,14 +420,14 @@ object DslSpec extends TestSuite {
     "support common data types" - {
 
       case class Foo(value: String)
-      case class Bar(value: String)
+      case class Bar(value: String, unit: Unit)
```
Comment (krzemin) [inline chimney/src/test/scala/io/scalaland/chimney/DslSpec.scala]: I suggest to keep these test as they are. Instead, I would add new test case "support Unit" where you construct simple example with unit field in target case class.

File: chimney/src/test/scala/io/scalaland/chimney/DslSpec.scala [hunk 2]
```diff
@@ -676,6 +676,14 @@ object DslSpec extends TestSuite {
 
         transform[String](polySource) ==> monoTarget
       }
+      
+      "automatically fill Unit parameters" - {
+        case class Foo(value: String)
+        case class Bar[T](value: String, poly: T)
+        type UnitBar = Bar[Unit]
+
+        Foo("test").transformInto[UnitBar] ==> Bar("test", ())
```
Comment (krzemin) [inline chimney/src/test/scala/io/scalaland/chimney/DslSpec.scala]: What about not using type alias, but `transformInto[Bar[Unit]]` directly?

File: chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala [hunk 3]
```diff
@@ -661,6 +661,14 @@ trait TransformerMacros extends TransformerConfiguration {
             None
           }
         }
+        .orElse {
+          val targetTypeIsUnit = target.tpe <:< typeOf[Unit]
+          if (targetTypeIsUnit) {
```
Comment (krzemin) [inline chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala]: seems like this type test could be inlined

**Comment by ldrygala** [comment]
thanks @krzemin for helping me with my first contribution 👌

**Pull request merged**